src/BaseCollection.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Friendica;
+
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class BaseCollection
+ *
+ * The Collection classes inheriting from this abstract class are meant to represent a list of database record.
+ * The associated model class has to be provided in the child classes.
+ *
+ * Collections can be used with foreach().
+ *
+ * @package Friendica
+ */
+abstract class BaseCollection implements \Iterator
```

Comment:
This is a amazing way for fetchin more model-entities and use them, but it still doesn't solve:
* Calling a model-method once for many entities (like `Introduction::ignore()`)
* Instancing the whole model for each selected entity (memory allocation)
* Hard binding a whole business logic of a model in one class (like `Item` or `Contact`)

But as said in the forum post, I won't stay in the way anymore, because you really want it to use models this way and I think at least I will live with it :-)

Reply:
No, I’m very interested in your high-level approach because:
1. It is unfamiliar to me
2. I may be missing something in the larger scheme of things. Sure, it would work in the cozy simple context of Introductions, but it could become spectacularly unwieldy with even mild complexity scenarios like 1-n relationships, even without mentioning n-m relationships with added data.
3. How would you handle those with Repositories and Model entities?

Reply:
Sorry for the harsh words .. I was in a bad mood yesterday :-( .. I'll try to formulate an approach which maybe fits for both of us